### PR TITLE
Change LOC branch for VS 17.3

### PR DESCRIPTION
**Pull request by JoeRobich**
Stop performing LOC on 17.2 branch and start localizing 17.3.

### CODE REVIEW

**Review by dibarbet — APPROVED** [review]

File: azure-pipelines-official.yml [hunk 1]
```diff
@@ -82,11 +82,11 @@ stages:
     demands: ImageOverride -equals Build.Windows.Amd64.VS2022
 
   jobs:
-  - ${{ if eq(variables['Build.SourceBranch'], 'refs/heads/release/dev17.2') }}:
+  - ${{ if eq(variables['Build.SourceBranch'], 'refs/heads/release/dev17.3') }}:
     - template: /eng/common/templates/job/onelocbuild.yml
       parameters:
         MirrorRepo: roslyn
-        MirrorBranch: release/dev17.2
+        MirrorBranch: release/dev17.3
```
Comment (dibarbet) [inline azure-pipelines-official.yml:89]: at what point do we update this in general?  Should it be done every time we create a new release branch?

**Comment by RikkiGibson** [comment]
What's inducing us to do this at this specific time? Just want to make sure we have it documented in the appropriate place to keep this updated, i.e. "When X happens, make sure you update the LOC branch".

**Comment by JoeRobich** [comment]
> What's inducing us to do this at this specific time?

@RikkiGibson Well, Richa sends an email when it is time to perform this task. I think perhaps the reason this is the right time is because we snapped for the last preview that should contain new features (and thusly new strings).

File: azure-pipelines-official.yml (hunk 1 above)
Comment (JoeRobich) [reply azure-pipelines-official.yml:89]: We localize one branch at a time and it requires that this stay in sync with internal localization configuration. I believe this change is typically requested after we've snapped for the last preview that will contain new features.

**Pull request merged**